pkg/core/bundle_unpack.go

```diff
@@ -13,14 +16,58 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-func unpackBundleDescriptor(ctx context.Context, bundle *Bundle) error {
-
-	bundleDescriptorBuffer, err := storage.ReadTee(ctx,
-		bundle.MetaStore, model.GetArchivePathToBundle(bundle.RepoID, bundle.BundleID),
-		bundle.ConsumableStore, model.GetConsumablePathToBundle(bundle.BundleID))
+func setBundleIDFromConsumableStore(ctx context.Context, bundle *Bundle) error {
+	// todo: use KeysPrefix() after that's implemented.
+	keys, err := bundle.ConsumableStore.Keys(ctx)
 	if err != nil {
 		return err
 	}
+	// todo: this is re-encoding model.GetConsumablePathToBundle()
```

Comment:
getting the bundle id out of the `.datamon/` folder on disk, and in particular the duplication (in the sense that there are two implementations that can't be changed independently) of this function with `model` is one of the remaining pain-points in this implementation.